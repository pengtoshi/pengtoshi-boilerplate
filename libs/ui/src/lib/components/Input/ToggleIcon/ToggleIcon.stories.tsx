import type { StoryFn } from "@storybook/react";
import { useState } from "react";
import type { ComponentProps } from "react";
import { ToggleIcon } from "./ToggleIcon";

export default { component: ToggleIcon };

const Template: StoryFn<ComponentProps<typeof ToggleIcon>> = (args: ComponentProps<typeof ToggleIcon>) => {
  const [active, setActive] = useState(false);
  return <ToggleIcon {...args} active={active} handleClick={() => setActive(!active)} />;
};

export const Default = Template.bind({});
Default.args = {
  name: "Download",
};
