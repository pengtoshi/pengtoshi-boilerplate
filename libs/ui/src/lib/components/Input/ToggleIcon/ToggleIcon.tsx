import clsx from "clsx";
import { useEffect, useState } from "react";
import type { UIProps } from "../../../props";
import { Interaction } from "../../Display/Interaction/Interaction";
import { Icon } from "../../Icon/Icon";
import type { IconProps } from "../../Icon/Icon";

export interface ToggleIconProps extends UIProps.Button {
  active: boolean;
  name: IconProps["name"];
  handleClick: () => void;
}

export const ToggleIcon = ({ active, handleClick, className, name, ...props }: ToggleIconProps) => {
  const [internalActive, setInternalActive] = useState(active);

  useEffect(() => {
    setInternalActive(active);
  }, [active]);

  return (
    <button
      type="button"
      className={clsx("group relative", className)}
      onClick={() => {
        setInternalActive(!internalActive);
        handleClick();
      }}
      {...props}
    >
      <Icon className={internalActive ? "text-primary-500" : "text-gray-400"} name={name} />
      <Interaction focus={false} className={clsx("-left-1 -top-1 h-8 w-8 rounded-full")} />
    </button>
  );
};
