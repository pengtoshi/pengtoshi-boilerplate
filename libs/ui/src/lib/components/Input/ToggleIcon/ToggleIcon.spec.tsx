import { composeStories } from "@storybook/react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./ToggleIcon.stories";

describe("ToggleIcon", () => {
  Object.entries(composeStories(stories)).forEach(([, Story]) => {
    it("should render", () => {
      const { baseElement } = render(<Story />);
      expect(baseElement).toMatchSnapshot();
    });
  });
});
