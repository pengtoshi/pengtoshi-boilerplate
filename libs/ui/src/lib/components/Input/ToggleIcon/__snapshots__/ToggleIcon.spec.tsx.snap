// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleIcon should render 1`] = `
<body>
  <div>
    <section
      aria-atomic="false"
      aria-label="Notifications alt+T"
      aria-live="polite"
      aria-relevant="additions text"
      tabindex="-1"
    />
    <button
      class="group relative"
      type="button"
    >
      <div
        class="text-gray-400"
        height="24"
        width="24"
      />
      <div
        class="absolute inset-0 bg-gray-950 opacity-0 transition-opacity duration-300 group-hover:opacity-[0.06] group-active:opacity-[0.12] -left-1 -top-1 h-8 w-8 rounded-full"
      />
    </button>
  </div>
</body>
`;
